lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -168,10 +186,71 @@ export abstract class CacheManager implements ICacheManager {
      * retrieve a credential - accessToken, idToken or refreshToken; given the cache key
      * @param key
      */
-    getCredential(key: string): CredentialEntity {
+    getCredential(key: string): CredentialEntity | null {
         return this.getItem(key, CacheSchemaType.CREDENTIAL) as CredentialEntity;
     }
 
+    /**
+     * Helper function to retrieve IdTokenEntity from cache
+     * @param clientId 
+     * @param account 
+     * @param inputRealm 
+     */
+    getIdTokenEntity(clientId: string, account: AccountInfo): IdTokenEntity | null {
+        const idTokenKey: string = CredentialEntity.generateCredentialCacheKey(
+            account.homeAccountId,
+            account.environment,
+            CredentialType.ID_TOKEN,
+            clientId,
+            account.tenantId
+        );
+
+        return this.getCredential(idTokenKey) as IdTokenEntity;
+    }
+
+    /**
+     * Helper function to retrieve AccessTokenEntity from cache
+     * @param clientId 
+     * @param account 
+     * @param scopes 
+     * @param inputRealm 
+     */
+    getAccessTokenEntity(clientId: string, account: AccountInfo, scopes: ScopeSet): AccessTokenEntity | null {
+        const accessTokenFilter: CredentialFilter = {
+            homeAccountId: account.homeAccountId,
+            environment: account.environment,
+            credentialType: CredentialType.ACCESS_TOKEN,
+            clientId,
+            realm: account.tenantId,
+            target: scopes.printScopesLowerCase()
```

Comment:
We are storing `scopes` in lower case and `look-up` in lower case. We need to address this, as I observed in cache-compat test cases, other libraries store the case as is. We may need to talk about this.

My take: Look up should be case insensitive but storing should be as is.

Reply:
100% agree. As this PR is scoped to a refactor, let's take functional changes in a follow up PR